component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/jwt/JWTBearerGrantHandler.java

```diff
@@ -471,7 +471,8 @@ protected void setAuthorizedUser(OAuthTokenReqMessageContext tokenReqMsgCtx, Ide
             authenticatedUser.setAuthenticatedSubjectIdentifier(authenticatedSubjectIdentifier);
         } else {
             authenticatedUser = AuthenticatedUser
-                    .createLocalAuthenticatedUserFromSubjectIdentifier(authenticatedSubjectIdentifier);
+                    .createFederateAuthenticatedUserFromSubjectIdentifier(authenticatedSubjectIdentifier);
+            authenticatedUser.setUserName(authenticatedSubjectIdentifier);
```

Comment:
Using createFederateAuthenticatedUserFromSubjectIdentifier will set the authenticatedUser AuthenticatedSubjectIdentifier to the given value. Since this is a federated user that should be sufficient and we don't need to set the userName.

Reply:
Username filed is not populated from createFederateAuthenticatedUserFromSubjectIdentifier method. OAuth2ServiceJWTGrantTestCase was failing without the username. Similar to the approach in the SSAML2BearerGrantHandler [1], we have to set username explicitly to avoid these failures.

[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java#L1275